src/StructuredLogger/BinaryLogger/BuildEventArgsReader.cs

```diff
@@ -250,23 +352,22 @@ private void ReadBlob(BinaryLogRecordKind kind)
                     prefixBytes = reader.ReadBytes(12 - bytesRead);
                 }
 
-                bytes = binaryReader.ReadBytes(length - prefixBytes.Length);
-
-                byte[] fullBytes = new byte[length];
-                prefixBytes.CopyTo(fullBytes, 0);
-                bytes.CopyTo(fullBytes, prefixBytes.Length);
-                bytes = fullBytes;
-
-                OnBlobRead?.Invoke(kind, bytes);
+                Stream prefixStream = new MemoryStream(prefixBytes);
+                Stream dataStream = binaryReader.BaseStream.Slice(length - prefixBytes.Length);
+                return prefixStream.Concat(dataStream);

```

Comment:
This is neat and if implemented right, also much cheaper than byte arrays. Just worried about an extra dependency.

I also wasn't happy when we added Microsoft.NET.StringTools.dll as a core dependency to MSBuild when I was not looking.

Reply:
Thank you for learning me something new with the source only packages! ;-)